tcms/testcases/tests/test_views.py

```diff
@@ -120,6 +120,84 @@ def test_clean(self):
         self.assertEqual(data, [])
 
 
+class TestNewCase(BasePlanCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.new_case_url = reverse('testcases-new')
+
+        cls.summary = 'summary'
+        cls.text = 'some text description'
+        cls.script = 'some script'
+        cls.arguments = 'args1, args2, args3'
+        cls.requirement = 'requirement'
+        cls.link = 'http://somelink.net'
+        cls.notes = 'notes'
+        cls.data = {
+            'summary': cls.summary,
+            'default_tester': cls.tester.pk,
+            'product': cls.case.category.product.pk,
+            'category': cls.case.category.pk,
+            'case_status': cls.case_status_confirmed.pk,
+            'priority': cls.case.priority.pk,
+            'text': cls.text,
+            'script': cls.script,
+            'arguments': cls.arguments,
+            'requirement': cls.requirement,
+            'extra_link': cls.link,
+            'notes': cls.notes
+        }
+
+        user_should_have_perm(cls.tester, 'testcases.add_testcase')
+
+    def test_create_test_case_successfully(self):
+        response = self.client.post(self.new_case_url, self.data)
+
+        test_case = TestCase.objects.get(summary='summary')
+        redirect_url = reverse('testcases-get', args=[test_case.pk])
+
+        self.assertRedirects(response, redirect_url)
+        self._assertTestCase(test_case)
+
+    def test_create_test_case_successfully_from_plan(self):
+        self.data['from_plan'] = self.plan.pk
+
+        response = self.client.post(self.new_case_url, self.data)
+
+        test_case = TestCase.objects.get(summary=self.summary)
+        redirect_url = "{0}?from_plan={1}".format(
+            reverse('testcases-get', args=[test_case.pk]), self.plan.pk)
+
+        self.assertRedirects(response, redirect_url)
+        self.assertEqual(test_case.plan.get(), self.plan)
+        self._assertTestCase(test_case)
```

Comment:
You will also have to assert that the test case was indeed added to the test plan. That is the test plans card displays the name of self.plan and links back to it.

Reply:
Should I assert that some piece of HTML is contained into the `response` or rather that the `test_case` is part `case`?

Reply:
Best to do both, that way you will catch rendering errors if we have some.

Reply:
I have asserted the objects.
However, on the client side the test plans are added to the page via js, so at this moment the `response` objects does not contain any plans and I cannot assert anything